app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt

```diff
@@ -102,29 +139,154 @@ class PlatformParameterIntegrationTest {
   }
 
   @Test
-  fun testSplashTestActivity_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
-    launch(SplashTestActivity::class.java).use {
+  fun testIntegration_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    launch(SplashTestActivity::class.java).use { it ->
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      it.onActivity { it.splashTestActivityPresenter.loadPlatformParameters() }
       testCoroutineDispatchers.runCurrent()
+
       assertThat(ShadowToast.getLatestToast()).isNull()
     }
   }
 
   @Test
-  fun testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible() {
+  fun testIntegration_updateEmptyDatabase_readDatabase_checkWelcomeMsgIsVisible() {
     platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
     testCoroutineDispatchers.runCurrent()
 
-    launch(SplashTestActivity::class.java).use {
+    launch(SplashTestActivity::class.java).use { it ->
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      it.onActivity { it.splashTestActivityPresenter.loadPlatformParameters() }
+      testCoroutineDispatchers.runCurrent()
+
+      assertThat(ShadowToast.getLatestToast()).isNotNull()
+      assertThat(ShadowToast.getTextOfLatestToast()).isEqualTo(SplashTestActivity.WELCOME_MSG)
+    }
+  }
+
+  @Test
+  fun testIntegration_executeSyncUpWorkerCorrectly_readDatabase_checkWelcomeMsgIsVisible() {
+    launch(SplashTestActivity::class.java).use { it ->
+      // Set up versionName to get correct network response from mock platform parameter service.
+      setUpApplicationForContext(MockPlatformParameterService.appVersionForCorrectResponse)
+      platformParameterController.updatePlatformParameterDatabase(listOf())
```

Comment:
Maybe I didnt completely understand -- why are we updating this here with an empty list? Dont we want the DB to have some initial values? Or if you are using the initial value as empty, lets instead have the initial value explicitly as false

Reply:
Yeah we can have a false value in the database which will be better proof that the values changed.